lib/core/src/test/java/io/atlasmap/builder/DefaultAtlasMappingBuilderTest.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.atlasmap.builder;
+
+import static io.atlasmap.api.AtlasConstants.DEFAULT_SOURCE_DOCUMENT_ID;
+import static io.atlasmap.api.AtlasConstants.DEFAULT_TARGET_DOCUMENT_ID;
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+
+import io.atlasmap.api.AtlasException;
+import io.atlasmap.core.BaseDefaultAtlasContextTest;
+import io.atlasmap.v2.AtlasMapping;
+import io.atlasmap.v2.FieldType;
+
+public class DefaultAtlasMappingBuilderTest extends BaseDefaultAtlasContextTest {
+
+    @Test
+    public void test() throws Exception {
+        DefaultAtlasMappingBuilder builder = new DefaultAtlasMappingBuilder() {
+            @Override
+            public void processMapping() {
+                try {
+                    read(DEFAULT_SOURCE_DOCUMENT_ID, "/f1")
+                        .write(DEFAULT_TARGET_DOCUMENT_ID, "/f2");
+                    readConstant("c3").writeProperty(DEFAULT_TARGET_DOCUMENT_ID, "p4");
+                    readProperty(DEFAULT_SOURCE_DOCUMENT_ID, "p5").write(DEFAULT_TARGET_DOCUMENT_ID, "/f6");
+                } catch (AtlasException e) {
+                    addAudit(e);
```

Comment:
It seems that we could make processMapping to rethrow this exception and handle it in DefaultAtlasMappingBuilder in 99% of cases...

Reply:
Good catch, thanks! will do.